java/core/src/main/java/com/lancedb/lance/ipc/LanceScanner.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.lancedb.lance.ipc;
+
+import com.lancedb.lance.Dataset;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Optional;
+import org.apache.arrow.c.ArrowArrayStream;
+import org.apache.arrow.c.ArrowSchema;
+import org.apache.arrow.c.Data;
+import org.apache.arrow.dataset.scanner.ScanTask;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.vector.ipc.ArrowReader;
+import org.apache.arrow.vector.types.pojo.Schema;
+
+/** Scanner over a Fragment. */
+public class LanceScanner implements org.apache.arrow.dataset.scanner.Scanner {
+  Dataset dataset;
+
+  ScanOptions options;
+
+  BufferAllocator allocator;
+
+  private long nativeScannerHandle;
+
+  private LanceScanner() {}
+
+  /**
+   * Create a Scanner.
+   *
+   * @param dataset the dataset to scan
+   * @param options scan options
+   * @param allocator allocator
+   * @return a Scanner
+   */
+  public static LanceScanner create(Dataset dataset, ScanOptions options,
+      BufferAllocator allocator) {
+    LanceScanner scanner = createScanner(dataset, Optional.empty(), options.getColumns(),
+        options.getSubstraitFilter(), options.getFilter(), options.getBatchSize());
+    scanner.allocator = allocator;
+    scanner.dataset = dataset;
+    scanner.options = options;
+    return scanner;
+  }
+
+  static native LanceScanner createScanner(Dataset dataset, Optional<Integer> fragmentId,
```

Comment:
Which method actually create a scanner with 1 fragment?

Reply:
This represents the FragmentScanner created from the DatasetFragment class

Reply:
The Rust scanner, fragments is  fragments: Option<Vec<Fragment>>,
https://github.com/lancedb/lance/blob/098f7308b628d71746a4363e5e8778ad1b2500cb/rust/lance/src/dataset/scanner.rs#L179
https://github.com/lancedb/lance/blob/098f7308b628d71746a4363e5e8778ad1b2500cb/rust/lance/src/dataset/scanner.rs#L219